src/mpid/ch4/netmod/include/netmod_am_fallback_send.h

```diff
@@ -25,7 +30,12 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_ssend(const void *buf,
                                                 MPIR_Comm * comm, int context_offset,
                                                 MPIDI_av_entry_t * addr, MPIR_Request ** request)
 {
-    return MPIDIG_mpi_ssend(buf, count, datatype, rank, tag, comm, context_offset, addr, request);
+    int mpi_errno;
+    int protocol = MPIDIG_AM_PROTOCOL__EAGER;
+
+    protocol = MPIDI_NM_am_choose_protocol(buf, count, datatype, 0, MPIDIG_SSEND_REQ);
```

Comment:
I still think it is weird to call an `MPIDI_NM` API here. Could the generic layer call this function at the top of `MPIDIG_mpi_ssend_new` for example? Or is there something I'm missing? IMO, this would make the abstraction a little cleaner.